### DISCUSSION TITLE
jboss-marshalling is deprecated and planned for removal

### DISCUSSION THREAD

**Discussion by TheKrizz** [keycloak/keycloak#31121] [Q&A] [answered]
If we start keycloak we get following log messages:
```
WARN  [org.infinispan.PERSISTENCE] (keycloak-cache-init) ISPN000554: jboss-marshalling is deprecated and planned for removal
INFO  [org.infinispan.CONTAINER] (keycloak-cache-init) ISPN000556: Starting user marshaller 'org.infinispan.jboss.marshalling.core.JBossUserMarshaller'
```

This is our cache-ispn.xml
<?xml version="1.0" encoding="UTF-8"?>

<infinispan
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="urn:infinispan:config:14.0 http://www.infinispan.org/schemas/infinispan-config-14.0.xsd"
        xmlns="urn:infinispan:config:14.0">

    <jgroups>
        <stack name="tcpping" extends="tcp">
            <TCP bind_port="${env.KC_BIND_PORT}" />
            <TCPPING initial_hosts="${env.KC_INITIAL_HOSTS}"
                port_range="0"
                stack.combine="REPLACE"
                stack.position="MPING" />
            <SYM_ENCRYPT keystore_name="${env.KC_ISPN_SECKEY_PATH}"
                keystore_type="PKCS12"
                store_password="${env.KC_ISPN_STOREPW}"
                key_password="${env.KC_ISPN_KEYPW}"
                alias="infinispan"
                stack.combine="INSERT_AFTER"
                stack.position="VERIFY_SUSPECT2"/>
        </stack>
    </jgroups>

    <cache-container name="keycloak">
        <transport cluster="${env.KC_CLUSTER_NAME}" lock-timeout="60000" stack="tcpping"/>
        <local-cache name="realms" simple-cache="true">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <memory max-count="10000"/>
        </local-cache>
        <local-cache name="users" simple-cache="true">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <memory max-count="10000"/>
        </local-cache>
        <distributed-cache name="sessions" owners="2">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <distributed-cache name="authenticationSessions" owners="2">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <distributed-cache name="offlineSessions" owners="2">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <distributed-cache name="clientSessions" owners="2">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <distributed-cache name="offlineClientSessions" owners="2">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <distributed-cache name="loginFailures" owners="2">
            <expiration lifespan="-1"/>
        </distributed-cache>
        <local-cache name="authorization" simple-cache="true">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <memory max-count="10000"/>
        </local-cache>
        <replicated-cache name="work">
            <expiration lifespan="-1"/>
        </replicated-cache>
        <local-cache name="keys" simple-cache="true">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <expiration max-idle="3600000"/>
            <memory max-count="1000"/>
        </local-cache>
        <distributed-cache name="actionTokens" owners="2">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <expiration max-idle="-1" lifespan="-1" interval="300000"/>
            <memory max-count="-1"/>
        </distributed-cache>
    </cache-container>

How can we fix both log entries?
I tried switching key media-type and value media-type to application/x-protostream
Also tried adding in cache-container something like 
```
<serialization marshaller="PROTOSTREAM">
```

So what is the solution?!

Keycloak Version is 23.06

**Comment by TheKrizz** [comment, marked as answer]
Fixed with newer Keycloak Version

**Reply by bsiara** [reply]
Hi, which marshaller you used for x-protostream encoding?
